### ISSUE TITLE
Where (or How) to get accuracy after each "epoch"?

### ISSUE THREAD

**Issue by fleapapa** [01org/daal#14]
I noticed that there is a "accuracyThreshold" field in source code of several optimization solvers' parameter set.

My question is: where or how to get or calculate the "accuracy" of a on-going learning model of a distributed neural net after each epoch or after a certain number of iterations from the learning model?

I need the accuracy information so that i can have the model stop learning once when it reaches a threshold of accuracy (but how to know this???) instead of waiting for a fix number of "epochs" which can take very long time.

Any pointer will be very appreciated!

**Comment by Vika-F** [comment]
Intel(R) DAAL provides the ability to get the value of the loss function for every batch.
And based on this value you can define any stopping criteria you like.
Please make the following modifications to the sample to access the value of the loss function.

First, save the identifier of the loss layer when initializing the topology of a network.
Please add following global variable into the sample:
```c++
LayerIds layerIds;
```
This variable will hold the identifiers of all layers in the topology.

Second, change the line that creates the topology by providing the storage for layers' identifiers:
```c++
/* Configure the neural network topology */
topology = configureNet(&layerIds);
```

And third, here is the code that prints the value of the loss function after compute() step on the local nodes.
```c++
/* Perform forward and backward pass through the neural network to compute weights and biases derivatives on worker node */
netLocal->compute();

/* Get the model of the neural network on local node */
training::ModelPtr trainingModel = netLocal->input.get(training::inputModel);
/* Get the result of the loss layer */
services::SharedPtr<layers::forward::Result> lossLayerResult = trainingModel->getForwardLayer(layerIds.sm)->getLayerResult();
/* Print the value computed by the loss layer */
printTensor(lossLayerResult->get(layers::forward::value), "loss value: ");
```

Regarding the "accuracyThreshold" parameter of the optimization solver.
The present version of the library does not use this parameter for the neural network training.
We investigate the options for adding support of accuracy threshold to the neural network training algorithm to have its behavior more configurable in the future releases of the library.

Best regards,
Victoriya

**Comment by fleapapa** [comment]
Thanks a lot for the tip!

Will an option for calculating one single loss on master node (ie., netMaster) be also investigated? The option can be a big convenience, especially for topologies with many worker nodes.
